File: backend/app/schemas/paper.py
from datetime import datetime

from pydantic import BaseModel


class PaperCreateFromLink(BaseModel):
    source: str


class PaperResponse(BaseModel):
    id: int
    title: str
    authors: str
    year: int | None
    abstract: str
    pdf_path: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/paper.ts
export interface Paper {
  id: number
  title: string
  authors: string
  year: number | null
  abstract: string
  pdf_path: string
  created_at: string
}

export interface Analysis {
  title: string
  authors: string
  year: number | null
  TLDR: string
  problem_gap: string
  motivation: string
  hypothesis: string
  methodology: string
  main_finding: string
  limitations: string
  confidence_score: number
}

export interface ChatMessage {
  role: 'user' | 'assistant'
  message: string
  citations?: number[]
}
